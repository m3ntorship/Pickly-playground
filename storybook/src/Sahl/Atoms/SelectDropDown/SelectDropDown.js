import React from "react";
import PropTypes from "prop-types";
// import ArrowDown from "../../assets/icons/DownArrow";

function SelectDropDown(props) {
  const { className, options, label } = props;
  return (
    <div className={`${className} flex items-center`}>
      <label
        htmlFor={label}
        className={`text-body-small text-dark-grey capitalize ${
          label ? "inline-block mr-2" : "hidden"
        }`}
      >
        {label}:
      </label>
      <select
        id={label}
        className={`appearance-none bg-white w-full m-0 font-sans text-grey text-sm capitalize cursor-pointer outline-none`}
      >
        {options.map((option, i) => (
          <option value={option} key={option + i}>
            {option}
          </option>
        ))}
      </select>
    </div>
  );
}

SelectDropDown.propTypes = {
  options: PropTypes.array.isRequired,
  className: PropTypes.string,
  label: PropTypes.string,
};

export default SelectDropDown;
