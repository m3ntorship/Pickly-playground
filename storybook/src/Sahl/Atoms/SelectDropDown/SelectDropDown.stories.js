import React from "react";

import SelectDropDown from "./SelectDropDown";

const defaultExport = {
  title: "Sahl/Atoms/SelectDropDown",
  component: SelectDropDown,
  // argTypes: {
  //   options: {
  //     control: {
  //       type: "select",
  //       options: ["Item One", "Item Two", "Item Three"],
  //     },
  //   },
  // },
};

export default defaultExport;

const Template = (args) => <SelectDropDown {...args} />;

export const Default = Template.bind({});
Default.args = {
  options: ["public", "friends", "custom"],
  label: "label",
  className: "",
};
