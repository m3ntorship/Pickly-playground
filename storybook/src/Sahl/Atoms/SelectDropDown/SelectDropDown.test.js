import React from "react";
import renderer from "react-test-renderer";
import SelectDropDown from "./SelectDropDown";

describe("SelectDropDown Component", () => {
  it("should render with passed options options=['Option 1','Option 2','Option 3']", () => {
    const tree = renderer
      .create(<SelectDropDown options={["Option 1", "Option 2", "Option 3"]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render without label by default", () => {
    const tree = renderer
      .create(<SelectDropDown options={["Option 1", "Option 2", "Option 3"]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with passed label", () => {
    const tree = renderer
      .create(
        <SelectDropDown
          options={["Option 1", "Option 2", "Option 3"]}
          label="testlabel"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
