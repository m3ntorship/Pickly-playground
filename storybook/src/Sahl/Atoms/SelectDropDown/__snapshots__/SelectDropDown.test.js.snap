// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectDropDown Component should render with passed label 1`] = `
<div
  className="undefined flex items-center"
>
  <label
    className="text-body-small text-dark-grey capitalize inline-block mr-2"
    htmlFor="testlabel"
  >
    testlabel
    :
  </label>
  <div
    className="flex items-center"
  >
    <select
      className="relative z-10 appearance-none border-none bg-transparent pr-4 m-0 w-full font-sans text-body-small capitalize cursor-pointer outline-none"
      id="testlabel"
    >
      <option
        value="Option 1"
      >
        Option 1
      </option>
      <option
        value="Option 2"
      >
        Option 2
      </option>
      <option
        value="Option 3"
      >
        Option 3
      </option>
    </select>
    <svg
      className="w-4 h-4  cursor-pointer -ml-3.5"
      fill="none"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16.293 9.29297L12 13.586L7.70697 9.29297L6.29297 10.707L12 16.414L17.707 10.707L16.293 9.29297Z"
        fill="#5B6366"
      />
    </svg>
  </div>
</div>
`;

exports[`SelectDropDown Component should render with passed options options=['Option 1','Option 2','Option 3'] 1`] = `
<div
  className="undefined flex items-center"
>
  <label
    className="text-body-small text-dark-grey capitalize hidden"
  >
    :
  </label>
  <div
    className="flex items-center"
  >
    <select
      className="relative z-10 appearance-none border-none bg-transparent pr-4 m-0 w-full font-sans text-body-small capitalize cursor-pointer outline-none"
    >
      <option
        value="Option 1"
      >
        Option 1
      </option>
      <option
        value="Option 2"
      >
        Option 2
      </option>
      <option
        value="Option 3"
      >
        Option 3
      </option>
    </select>
    <svg
      className="w-4 h-4  cursor-pointer -ml-3.5"
      fill="none"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16.293 9.29297L12 13.586L7.70697 9.29297L6.29297 10.707L12 16.414L17.707 10.707L16.293 9.29297Z"
        fill="#5B6366"
      />
    </svg>
  </div>
</div>
`;

exports[`SelectDropDown Component should render without label by default 1`] = `
<div
  className="undefined flex items-center"
>
  <label
    className="text-body-small text-dark-grey capitalize hidden"
  >
    :
  </label>
  <div
    className="flex items-center"
  >
    <select
      className="relative z-10 appearance-none border-none bg-transparent pr-4 m-0 w-full font-sans text-body-small capitalize cursor-pointer outline-none"
    >
      <option
        value="Option 1"
      >
        Option 1
      </option>
      <option
        value="Option 2"
      >
        Option 2
      </option>
      <option
        value="Option 3"
      >
        Option 3
      </option>
    </select>
    <svg
      className="w-4 h-4  cursor-pointer -ml-3.5"
      fill="none"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16.293 9.29297L12 13.586L7.70697 9.29297L6.29297 10.707L12 16.414L17.707 10.707L16.293 9.29297Z"
        fill="#5B6366"
      />
    </svg>
  </div>
</div>
`;
